Assets/Scripts/Minigames/CowardMiniGame/GearBehavior.cs

```diff
@@ -1,114 +1,134 @@
 
     [Header("Individual Gear")]
-    [SerializeField] private GameObject[] _gearSize;
-    [SerializeField] private GameObject _gearIndi;
-    private int _gearSizeNum;
-    //private bool _scrollable;
-    private bool _doOnce;
-    //private PlayerController _pc;
-    private bool _interact;
+    [SerializeField] private GameObject[] _gearSizes;
+    [SerializeField] private int _startingGearIndex;
+    private int _currentGearSizeIndex;
 
     [Header("Correct Gear")]
     [SerializeField] private int _rightGearNum;
-    [SerializeField] private Color _matRed;
-    [SerializeField] private Color _matGreen;
-    private Renderer _rndr;
-
-    // Start is called before the first frame update
-    void Start()
-    {
-        _doOnce = true;
-        _gearSizeNum = 1;
-        //_scrollable = false;
-        //GameObject _pcObject = GameObject.FindWithTag("Player");
-        //_pc = _pcObject.GetComponent<PlayerController>();
-        _rndr = this.GetComponent<Renderer>();
-    }
+    [SerializeField] private float _rotationSpeed;
+    private Vector3 _rotationAngle;
+    private bool _isComplete = false;
+    public bool IsComplete { get => _isComplete; private set => _isComplete = value; }
 
     /// <summary>
-    /// used for gradually cycling through gear sizes through the list of gears
+    /// Ensures only starting gear is active
     /// </summary>
-    void FixedUpdate()
+    void Awake()
     {
-        transform.localScale = new Vector3(1f, _gearIndi.transform.localScale.y + 0.5f, 1f);
-        if (_interact && _gearSizeNum != _gearSize.Length && _doOnce == true)// && _scrollable == true)
+        // Ensures rotation is always clockwise
+        if (_rotationSpeed > 0)
         {
-            _gearSizeNum++;
-            _doOnce = false;
-            StartCoroutine(doOnceCooldown());
+            _rotationAngle = new Vector3(0, 0, -_rotationSpeed);
         }
-        if (_interact && _gearSizeNum == _gearSize.Length && _doOnce == true)// && _scrollable == true)
+        else
+        {
+            _rotationAngle = new Vector3(0, 0, _rotationSpeed);
+        }
+
+        if (_startingGearIndex >= _gearSizes.Length || _startingGearIndex < 0)
         {
-            _gearSizeNum = 1;
-            _doOnce = false;
-            StartCoroutine(doOnceCooldown());
+            _startingGearIndex = 0;
+        }
+        _currentGearSizeIndex = _startingGearIndex;
+
+        for (int i = 0; i < _gearSizes.Length; ++i)
+        {
+            if (i == _startingGearIndex)
+                _gearSizes[i].SetActive(true);
+            else
+                _gearSizes[i].SetActive(false);
         }
-        _gearIndi = _gearSize[_gearSizeNum - 1];
-        //Debug.Log(_gearSizeNum);
     }
+
     /// <summary>
-    /// if right gear number then make green if not keep red
+    /// Rotates gear if it's correctly slotted
     /// </summary>
-    void Update()
+    private void FixedUpdate()
```

Comment:
It may be worth changing this to a coroutine that waits for fixed update. So the check can start when the minigame is started and stop after the it is completed